samples/cellular/nrf_cloud_multi_service/src/sample_reboot.c

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/reboot.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <modem/lte_lc.h>
+
+#include "sample_reboot.h"
+
+#define ERROR_REBOOT_S		30
+#define NORMAL_REBOOT_S		10
+
+LOG_MODULE_REGISTER(sample_reboot, CONFIG_MULTI_SERVICE_LOG_LEVEL);
+
+/* Called by various subsystems in the sample when a clean reboot is needed. */
+void sample_reboot(const bool error)
+{
+	unsigned int delay_s = error ? ERROR_REBOOT_S  : NORMAL_REBOOT_S;
```

Comment:
While this is simpler than having a time parameter, it also makes it less flexible.  Why did you chose this direction?

Reply:
I no longer recall the specifics of how I arrived at this. The history is complicated because you and I both simultaneously refactored the reboot logic in essentially opposite directions, and I had to merge those back into something congruent.

I think I found the various configs you had for different reboot types confusing, and wanted to simplify down to just two types of reboot (error vs. normal). And after that, there didn't seem much point in offering delay customization, so I just removed it altogether.

I could add that back in if you want. I think there is a small risk the additional feature might be a red herring for beginners trying to understand the code. But I doubt it would have much impact either way.

I'll add it back in on my next rebase unless you change your mind.

Reply:
The various values came from someone else. I was just keeping things the same, with the assumption that there was some reason the existing values were selected. Was the 5 second delay not used at all?

Reply:
AFAIK, there is no technical reason to have customizable reboot timings, no.

From a UX perspective, it makes sense for reboots with critical information (an error message) to have a long delay, and for automatic/normal reboots to go by quickly.

I'm guessing that is where the 5 second delay came from, but I don't know the specific source you are referencing, so I can't know for sure what the reason there was.

But I do know it has no impact on functionality.

Reply:
The 5 second value was defined in fota_support.h:
#define FOTA_REBOOT_DELAY_S	5

Perhaps no one uses it.

Reply:
Yeah I think that is one of the extra values that I found confusing. I don't know where it originated before it was placed in `fota_support.h`. I can trace that origin if you would like, but I am certain there are no circumstances where it is important that the reboot delay specifically be 5 seconds, and currently no one is using the value.

It would seem I forgot to remove it from `fota_support.h`, so, my bad. I'll do that in the next rebase.

Reply:
discussed offline